src/ripple/overlay/impl/ProtocolMessage.h

```diff
@@ -120,51 +120,94 @@ buffersBegin(BufferSequence const& bufs)
         bufs);
 }
 
+/** Parse a message header
+ * @return a seated optional if the message header was successfully
+ *         parsed. An unseated optional otherwise, in which case
+ *         @param ec contains more information:
+ *         - set to `errc::success` if not enough bytes were present
+ *         - set to `errc::no_message` if a valid header was not present
+ */
 template <class BufferSequence>
 boost::optional<MessageHeader>
-parseMessageHeader(BufferSequence const& bufs, std::size_t size)
+parseMessageHeader(
```

Comment:
What do you think of putting the bitwise comparison inside of a named lambda, where the name describes what is being checked? I don't really like bitwise operations in a conditional directly, because it doesn't read that easily. It always takes me a second to figure out whats going on, and I think sometimes subtle bugs can occur. Remember this: https://github.com/ripple/rippled/issues/3417? At the least, I think we should add a comment saying exactly what is being checked.